x-pack/plugins/security/server/deprecations/kibana_dashboard_only_role.test.ts

```diff
@@ -0,0 +1,322 @@
+  let mockDeps: ReturnType<typeof getDepsMock>;
+  let mockContext: ReturnType<typeof getContextMock>;
+  let deprecationHandler: RegisterDeprecationsConfig;
+  beforeEach(() => {
+    mockContext = getContextMock();
+    mockDeps = getDepsMock();
+    registerKibanaDashboardOnlyRoleDeprecation(mockDeps);
+
+    expect(mockDeps.deprecationsService.registerDeprecations).toHaveBeenCalledTimes(1);
+    deprecationHandler = mockDeps.deprecationsService.registerDeprecations.mock.calls[0][0];
+  });
+
+  it('does not return any deprecations if security is not enabled', async () => {
+    mockDeps.license.isEnabled.mockReturnValue(false);
+
+    await expect(deprecationHandler.getDeprecations(mockContext)).resolves.toEqual([]);
+    expect(mockContext.esClient.asCurrentUser.security.getUser).not.toHaveBeenCalled();
+    expect(mockContext.esClient.asCurrentUser.security.getRoleMapping).not.toHaveBeenCalled();
+  });
+
+  it('does not return any deprecations if none of the users and role mappings has a dashboard only role', async () => {
+    mockContext.esClient.asCurrentUser.security.getUser.mockResolvedValue(
+      securityMock.createApiResponse({ body: { userA: createMockUser() } })
+    );
+
+    mockContext.esClient.asCurrentUser.security.getRoleMapping.mockResolvedValue(
+      securityMock.createApiResponse({
+        body: {
+          mappingA: { enabled: true, roles: ['roleA'], rules: {}, metadata: {} },
+        },
+      })
+    );
+
+    await expect(deprecationHandler.getDeprecations(mockContext)).resolves.toEqual([]);
+  });
+
+  it('returns deprecations even if cannot retrieve users due to permission error', async () => {
+    mockContext.esClient.asCurrentUser.security.getUser.mockRejectedValue(
+      new errors.ResponseError(securityMock.createApiResponse({ statusCode: 403, body: {} }))
+    );
+    mockContext.esClient.asCurrentUser.security.getRoleMapping.mockResolvedValue(
+      securityMock.createApiResponse({ body: { mappingA: createMockRoleMapping() } })
+    );
+
+    await expect(deprecationHandler.getDeprecations(mockContext)).resolves.toMatchInlineSnapshot(`
+            Array [
+              Object {
+                "correctiveActions": Object {
+                  "manualSteps": Array [
+                    "Make sure you have a \\"manage_security\\" cluster privilege assigned.",
+                  ],
+                },
+                "deprecationType": "feature",
+                "documentationUrl": "https://www.elastic.co/guide/en/kibana/some-branch/xpack-security.html#_required_permissions_7",
+                "level": "fetch_error",
+                "message": "You do not have enough permissions to fix this deprecation.",
+                "title": "The \\"kibana_dashboard_only_user\\" role is removed",
+              },
+            ]
+          `);
+  });
+
+  it('returns deprecations even if cannot retrieve users due to unknown error', async () => {
```

Comment:
❤️  I was worried that an error would prevent UA from surfacing the other deprecations, but this test assuages that concern.